srcts/src/components/card.ts

```diff
@@ -0,0 +1,277 @@
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    CLASS_CARD: "bslib-card",
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    CLASS_FULL_SCREEN: "bslib-full-screen",
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    CLASS_HAS_FULL_SCREEN: "bslib-has-full-screen",
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    CLASS_FULL_SCREEN_ENTER: "bslib-full-screen-enter",
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    CLASS_FULL_SCREEN_EXIT: "bslib-full-screen-exit",
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    ID_FULL_SCREEN_OVERLAY: "bslib-full-screen-overlay",
+  };
+
+  /**
+   * A Shiny-specific resize observer that ensures Shiny outputs in the main
+   * content areas of the sidebar resize appropriately.
+   * @private
+   * @type {ShinyResizeObserver}
+   * @static
+   */
+  private static shinyResizeObserver = new ShinyResizeObserver();
+
+  constructor(el: HTMLElement) {
+    // remove initialization attribute and script
+    el.removeAttribute(Card.attr.ATTR_INIT);
+    el.querySelector<HTMLScriptElement>(
+      `script[${Card.attr.ATTR_INIT}]`
+    )?.remove();
+
+    this.container = el;
+    Card.instanceMap.set(el, this);
+
+    // Let Shiny know to trigger resize when the card size changes
+    // TODO: shiny could/should do this itself (rstudio/shiny#3682)
+    Card.shinyResizeObserver.observe(this.container);
+
+    this._addEventListeners();
+    this._enableTooltips();
+    this.overlay = this._createOverlay();
+  }
+
+  enterFullScreen(event?: Event): void {
+    if (event) event.preventDefault();
+
+    this.overlay.container.addEventListener("click", () =>
+      this.exitFullScreen()
+    );
+    document.addEventListener(
+      "keyup",
+      (ev) => this._exitFullScreenOnEscape(ev),
+      false
+    );
+
+    const focusableElements = getAllFocusableChildren(this.container);
+    if (focusableElements.length > 0) {
+      // set focus on first focusable element in the card
+      focusableElements[0].focus();
+      // store the last focusable element so we can cycle focus
+      this.lastFocusInterior = focusableElements[focusableElements.length - 1];
+    } else {
+      // this card doesn't have any focusable elements, so focus is vaguely
+      // within the card (having clicked the full screen button). We're can't
+      // know exactly where focus is located (we've hidden the button), so we
+      // attach a listener to the next Tab keydown event to entrap focus within
+      // the full screen card.
+      this.lastFocusInterior = this.overlay.anchor;
+      if (!this.container.contains(document.activeElement)) {
+        this.prevFocusExterior = document.activeElement as HTMLElement;
+        this.prevFocusExterior.addEventListener(
+          "keydown",
+          (ev) => this._entrapFocus(ev),
+          { once: true }
+        );
+      }
+    }
```

Comment:
I'm not sure I fully understand the context here, but it smells like the current code is more complex than it needs to be, especially if we're willing to bring focus to the anchor? Note that, if we do this, I think we could also get rid of the `prevFocusExterior` field?

```suggestion
      this.overlay.anchor.focus();
      this.lastFocusInterior = this.overlay.anchor;
```

Reply:
We're going to go with `tabindex="-1"` and move focus to the card container. In upcoming work we'll revisit the expand button.

Reply:
I implemented the change to move focus to the card on fullscreen, which also required a small bit of logic to skip the close button on first tab (to move into the card rather than hit the close button first).

I decided not to modify the focus outline at all. Bootstrap (5 at least) uses `:focus-visible`, so the focus ring is shown only when the user has been tabbing to move focus. I think we should keep the focus ring in this case.